src/components/help/ProductTour.js

```diff
@@ -0,0 +1,65 @@
+import React, { useState } from "react";
+import dynamic from "next/dynamic";
+import { useTranslation } from "i18n";
+
+import { getSteps } from "./steps";
+
+import { useMediaQuery, useTheme } from "@material-ui/core";
+
+// only new users
+const Joyride = dynamic(() => import("react-joyride"));
+
+export default function ProductTour(props) {
+    const { onTourExit, runTour } = props;
+    const theme = useTheme();
+    const [tourStepIndex, setTourStepIndex] = useState(0);
+    const { t: i18nTour } = useTranslation("intro");
+    const isMobile = useMediaQuery(theme.breakpoints.down("xs"));
+
+    const handleJoyrideCallback = (callbackData) => {
+        const realCallback = async (callbackData) => {
+            const { action, index, type, status } = callbackData;
+            const { ACTIONS, EVENTS, STATUS } = await import("react-joyride");
+
+            if ([STATUS.FINISHED, STATUS.SKIPPED].includes(status)) {
+                setTourStepIndex(0);
+                onTourExit();
+            } else if (
+                [EVENTS.STEP_AFTER, EVENTS.TARGET_NOT_FOUND].includes(type)
+            ) {
+                const stepIndex = index + (action === ACTIONS.PREV ? -1 : 1);
+                setTourStepIndex(stepIndex);
+            }
+        };
+        realCallback(callbackData);
+    };
+
+    return (
+        <Joyride
+            steps={getSteps(i18nTour, isMobile)}
```

Comment:
I refactored code to split out `ProductTour` into its own component.  But that also made the tour code complicated in sense that we have to do a separate (smaller) tour for mobile screens because of the sidebar status (`open` prop).